ledger/complete/wal/checkpoint_v6_test.go

```diff
@@ -276,6 +276,52 @@ func TestCheckpointV6IsDeterminstic(t *testing.T) {
 	})
 }
 
+func TestWriteAndReadCheckpointV6LeafEmptyTrie(t *testing.T) {
+	unittest.RunWithTempDir(t, func(dir string) {
+		tries := []*trie.MTrie{trie.NewEmptyMTrie()}
+		fileName := "checkpoint-empty-trie"
+		logger := unittest.Logger()
+		require.NoErrorf(t, StoreCheckpointV6Concurrently(tries, dir, fileName, &logger), "fail to store checkpoint")
+		resultChan, err := OpenAndReadLeafNodesFromCheckpointV6(dir, fileName, &logger)
+		require.NoErrorf(t, err, "fail to read checkpoint %v/%v", dir, fileName)
+		for readResult := range resultChan {
```

Comment:
we'd better assert resultChan has exactly one result by keeping track of a `nResult`, and assert it equals `1`
```suggestion
		for readResult := range resultChan {
```